src/Formats/PrettyFormatHelpers.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Formats/PrettyFormatHelpers.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <Processors/Chunk.h>
+#include <Common/formatReadable.h>
+
+
+namespace DB
+{
+
+void writeReadableNumberTipIfSingleValue(WriteBuffer & out, const Chunk & chunk, const FormatSettings & settings, bool color)
+{
+    if (chunk.getNumRows() == 1 && chunk.getNumColumns() == 1)
+        writeReadableNumberTip(out, *chunk.getColumns()[0], 0, settings, color);
+}
+
+void writeReadableNumberTip(WriteBuffer & out, const IColumn & column, size_t row, const FormatSettings & settings, bool color)
+{
+    if (column.isNullAt(row))
+        return;
+
+    auto value = column.getFloat64(row);
+    auto threshold = settings.pretty.output_format_pretty_single_large_number_tip_threshold;
+
+    if (threshold && isFinite(value) && abs(value) > threshold)
+    {
+        if (color)
+            writeCString("\033[90m", out);
```

Comment:
These values are spooky if you don't interact with raw tty every day. Maybe use `const`s to store them?